README.md

```diff
@@ -1 +1,66 @@
 # snare
+
+`snare` is a simple GitHub webhooks runner. When a request comes in, it runs an
+arbitrary program for that repository informing it of the event -- that program
+can then perform whatever actions it wants.
+
+
+## Basic usage
+
+`snare` requires three command-line arguments to be specified:
+
+```
+Usage: snare -p <port> -r <repo scripts dir> -s <secret>
+```
+
+where:
+
+ * `<port>` is a port number (e.g. 4567)
+ * `<repo scripts dir>` is the directory where programs for each repository
+   targeted by the webhook are stored. For a repository `repo` owned by `user`
+   the command `<repo scripts dir>/<user>_<repo> <event> <path to GitHub JSON>`
+   will be run.
+ * `<secret>` is the GitHub secret which guarantees that hooks are coming from
+   your GitHub repository and not a malfeasant.
+
+The most important part of this are the per-repo programs. For example,
```

Comment:
This is the first mention of "per-repo programs". Is it the same as "repo scripts dir"?

Reply:
I agree that tidying this up would be good.